include/nbl/builtin/hlsl/emulated_float64_t.hlsl

```diff
@@ -0,0 +1,118 @@
+
+using float32_t = float;
+//using emulated_float64_t = double;
+
+namespace emulated
+{
+    struct emulated_float64_t
+    {
+        // TODO: change to `uint64_t` when on the emulation stage
+        using storage_t = float32_t;
+
+        storage_t data;
+
+        // constructors
+        // TODO: specializations?
+        template <typename T>
+        static emulated_float64_t create(T val)

```

Comment:
yes please, `float32_t` and `float16_t`, then default specialization should be the integers

Reply:
scratch that... overloads